CHANGELOG.md

```diff
@@ -7,6 +7,36 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Added
+
+- **Step resources:**
+  - Added a `step_resources` parameter to the `Step` class which should be used to describe the computational resources required to run a step.
+    `Executor` implementations can use this information. For example, if your step needs 2 GPUs, you should set
+    `step_resources=StepResources(gpu_count=2)` (`"step_resources": {"gpu_count": 2}` in the configuration language).
+  - Added a `Step.resources()` property method. By default this returns the value specified by the `step_resources` parameter.
+    If your step implementation always requires the same resources, you can just override this method so you don't have to provide
+    the `step_resources` parameter.
+- **Step execution:**
+  - Added an `executor` field to the `tango.yml` settings. You can use this to define the executor you want to use by default.
+  - Added a Beaker `Executor` to the Beaker integration, registered as an `Executor` with the name "beaker".
+    To use this executor, add these lines to your `tango.yml` file:
+    ```yaml
+    executor:
+      type: beaker
+      beaker_workspace: ai2/my-workspace
+      clusters:
+        - ai2/general-cirrascale
+    ```
+    See the docs for the `BeakerExecutor` for more information on the input parameters.
+
+### Changed
+
+- **CLI:**
+  - The `tango run` command will throw an error if you have uncommitted changes in your repository, unless
```

Comment:
Can we do this only if you use the Beaker executor? I think that's a pretty big obstacle for rapid development for many people. Constraining people's workflow always makes the library feel heavy.

Reply:
> Can we do this only if you use the Beaker executor?

I guess, but I don't think this will constrain anyone's workflow. Let's say you have some uncommitted changes and go:

```bash
> tango run ...
ERROR: You have uncommitted changes! Use --allow-dirty to force.
```

At first you're like "what the heck, that's annoying!" But it makes you pause and think "should I actually commit these changes?" If the answer to that is still "no", then you just hit the up-arrow key and append `--allow-dirty` to the last command. That's it, all good. Next time you hit the up-arrow for `tango run ...` again it will already have the `--allow-dirty` flag. So, at worst, it's an extremely minor inconvenience that costs you 3-5 seconds depending on how quickly you type, at best it saves you from not knowing what state your code was actually in when you ran that experiment.

Also: we save the `git` state of the repository in each `StepInfo` object (an important part of the user-facing API of Tango), so I think it's important that we nudge users to have a clean, committed working directory when they run steps, otherwise the `git` state in their `StepInfo` objects will be incorrect.